src/Core/Security/ProviderConfiguration/Infrastructure/Repository/DbReadConfigurationRepository.php

```diff
@@ -0,0 +1,250 @@
+                    $json['password_security_policy']['has_special_characters'],
+                    $json['password_security_policy']['can_reuse_passwords'],
+                    $json['password_security_policy']['attempts'],
+                    $json['password_security_policy']['blocking_duration'],
+                    $json['password_security_policy']['password_expiration_delay'],
+                    $excludedUserAliases,
+                    $json['password_security_policy']['delay_before_new_password'],
+                );
+
+                return new LocalCustomConfiguration($securityPolicy);
+
+            case Provider::OPENID:
+                $jsonSchemaValidatorFile = __DIR__ . '/../OpenId/Repository/CustomConfigurationSchema.json';
+                $json = $configuration->getJsonCustomConfiguration();
+                $this->validateJsonRecord($json, $jsonSchemaValidatorFile);
+                $jsonDecoded = json_decode($json, true);
+                $jsonDecoded['contact_template'] = $jsonDecoded['contact_template_id'] !== null
+                    ? $this->readOpenIdConfigurationRepository->getContactTemplate($jsonDecoded['contact_template_id'])
+                    : null;
+                $jsonDecoded['contact_group'] = $jsonDecoded['contact_group_id'] !== null
+                    ? $this->readOpenIdConfigurationRepository->getContactGroup($jsonDecoded['contact_group_id'])
+                    : null;
+                $jsonDecoded['authorization_rules'] =
+                    $this->readOpenIdConfigurationRepository->getAuthorizationRulesByConfigurationId(
+                        $configuration->getId()
+                    );
+
+                return new OpenIdCustomConfiguration($jsonDecoded);
+            case Provider::WEB_SSO:
+                $jsonSchemaValidatorFile = __DIR__ . '/../WebSSO/Repository/CustomConfigurationSchema.json';
+                $json = $configuration->getJsonCustomConfiguration();
+                $this->validateJsonRecord($json, $jsonSchemaValidatorFile);
+                $json = json_decode($json, true);
+
+                return new WebSSOCustomConfiguration(
+                    $json['trusted_client_addresses'],
+                    $json['blacklist_client_addresses'],
+                    $json['login_header_attribute'],
+                    $json['pattern_matching_login'],
+                    $json['pattern_replace_login']
+                );
+            default:
+                throw new \Exception("unknown configuration name, can't load custom config");
+        }
+    }
+
+    /**
+     * @param string $providerName
+     * @return Configuration
+     */
+    private function loadConfigurationByName(string $providerName): Configuration
+    {
+        $query = $this->translateDbName(
+            sprintf("SELECT *
+                FROM `:db`.`provider_configuration`
+                WHERE `name` = '%s'", $providerName)
```

Comment:
can you bind value instead of use sprintf

Reply:
This request is not  not exposed, just used internally